md-rest-api/src/main/java/com/wipro/ats/bdre/md/rest/ProcessAPI.java

```diff
@@ -830,14 +900,19 @@ RestWrapper executeProcess(@ModelAttribute("process")
                     fPid.setAccessible(true);
                 }
                 executionInfo.setOSProcessId(fPid.getInt(osProcess));
-                LOGGER.debug(" OS process Id : "+executionInfo.getOSProcessId() + "executed by " + principal.getName());
+                LOGGER.debug(" OS process Id : " + executionInfo.getOSProcessId() + "executed by " + principal.getName());
             } catch (Exception e) {
                 executionInfo.setOSProcessId(-1);
                 LOGGER.error(e + " Setting OS Process ID failed " + executionInfo.getOSProcessId());
             }
             restWrapper = new RestWrapper(executionInfo, RestWrapper.OK);
-        } catch (Exception e) {
-            LOGGER.error( e + " Executing workflow failed " +e.getCause());
+        } catch (MetadataException e) {
+            LOGGER.error(e + " Executing workflow failed " + e.getCause());
+            restWrapper = new RestWrapper(e.getMessage(), RestWrapper.ERROR);
+        }catch (SecurityException e) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS1166)